crates/router/src/connector/payu/transformers.rs

```diff
@@ -0,0 +1,550 @@
+use error_stack::ResultExt;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    core::errors,
+    pii::PeekInterface,
+    types::{self, api, storage::enums},
+    utils::OptionExt,
+};
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuPaymentsRequest {
+    customer_ip: String,
+    merchant_pos_id: String,
+    total_amount: String,
+    currency_code: String,
+    description: String,
+    pay_methods: PayuPaymentMethod,
+    continue_url: Option<String>,
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuPaymentMethod {
+    pay_method: PayuPaymentMethodData,
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum PayuPaymentMethodData {
+    Card(PayuCard),
+    Wallet(PayuWallet),
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum PayuCard {
+    #[serde(rename_all = "camelCase")]
+    Card {
+        number: String,
+        expiration_month: String,
+        expiration_year: String,
+        cvv: String,
+    },
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuWallet {
+    pub value: String,
+    #[serde(rename = "type")]
+    pub typo: String,
```

Comment:
Please name the field `wallet_type` or something similar. Using `typo` as a field name provides no information whatsoever.
Also, update this field to be an enum instead.